### PR TITLE
Data pages: improve citations

### PR DESCRIPTION
**Pull request by danyx23**
This PR overhauls how we do citations on datapages. Max and Joe came up with a new way for how to cite data pages themselves (the last thing in the citation section) that needs the citation information of the related topic page. 

There are also various minor improvements to the display of sources/origins.

This is related to #2755 and fixes these points:
* [x] Change the wording of "processed" or "adapated" by under the chart to "with minor/major processing by Our World In Data"
* [x] Switch to new data page citation format described by Max and Joe

### CODE REVIEW

**Comment by danyx23** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2786** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2786" target="_blank">Graphite</a> 
    * **PR #2798** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2798" target="_blank">Graphite</a> 
      * **PR #2799** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2799" target="_blank">Graphite</a> 
        * **PR #2800** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2800" target="_blank">Graphite</a> 
          * **PR #2801** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2801" target="_blank">Graphite</a> 
            * **PR #2802** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2802" target="_blank">Graphite</a> 
              * **PR #2806** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2806" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2806?utm_source=stack-comment).

**Review by sophiamersmann — APPROVED** [review]
Looks good to me! (though I didn't fully grasp what's grabbed from the topic page)

File: site/DataPageV2Content.tsx [hunk 1]
```diff
@@ -141,29 +152,41 @@ export const DataPageV2Content = ({
...
-        datapageData.origins[0].dateAccessed
-            ? dayjs(datapageData.origins[0].dateAccessed).format("MMMM D, YYYY")
-            : ""
-    const urlAccessed =
-        datapageData.origins &&
-        datapageData.origins.length &&
-        datapageData.origins[0].urlDownload
-    const citationLong = `${citationShort}. ${datapageData.title}. ${originsLong}, ${processedAdapted} by Our World In Data. Retrieved ${dateAccessed} from ${urlAccessed}`
+    const citationShort = `${attributionPotentiallyShortened} – with ${processedAdapted} by Our World In Data (${yearOfUpdate})`
+    const citationLonger = `${attributionPotentiallyShortened} – with ${processedAdapted} by Our World In Data (${yearOfUpdate})`
```
[the remark this answers is not in the document]
Comment (danyx23) [reply site/DataPageV2Content.tsx:181]: Ah good catch, thanks! I fixed this in a later Pr where I already have an intermediate variable that is useful for this.

**Comment by danyx23** [comment]
### Merge activity

* **Oct 24, 12:30 PM**: @danyx23 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2806).

**Pull request merged**